pandas/io/xml.py

```diff
@@ -98,17 +103,17 @@ class _XMLFrameParser:
 
     def __init__(
         self,
-        path_or_buffer,
-        xpath,
-        namespaces,
-        elems_only,
-        attrs_only,
-        names,
-        encoding,
-        stylesheet,
+        path_or_buffer: FilePath | ReadBuffer[bytes] | ReadBuffer[str],
+        xpath: str,
```

Comment:
The doc-strings says "str or regex" (maybe `str | re.Match`)

Reply:
I think this means that the string might be interpreted as regex, because passing in an actual regex raises